stl/inc/memory

```diff
@@ -308,6 +308,21 @@ _NoThrowFwdIt uninitialized_fill_n(_NoThrowFwdIt _First, const _Diff _Count_raw,
 }
 #endif // _HAS_IF_CONSTEXPR
 
+#if _HAS_CXX20
+template <class _Ty, class... _Types>
+_Ty* construct_at(_Ty* const _Location, _Types&&... _Args) {
+    return ::new (const_cast<void*>(static_cast<const volatile void*>(_Location))) _Ty(_STD forward<_Types>(_Args)...);
+}
+
+namespace ranges {
+    template <class _Ty, class... _Types>
+    _Ty* construct_at(_Ty* const _Location, _Types&&... _Args) {

```

Comment:
@CaseyCarter given that this is identical to the `std::` one could we get away with "using _STD construct_at`?

Reply:
I've been trying to decide if we should submit an LWG issue to remove it - I don't see the point of having two literally identical functions in different namespaces in the Standard LIbrary. It made sense to me when I said "we should add a `ranges::construct_at` that's constrained with `constructible_from`" but somewhere along the way that constraint disappeared.

Reply:
I think it's too late now that C++20 is out the door :/

Reply:
> It made sense to me when I said "we should add a `ranges::construct_at` that's constrained with `constructible_from`"

Why not just constrain the `std::` version?

Reply:
> Why not just constrain the `std::` version?

Mostly to avoid FUD around concepts in namespace `std`. LWG is conservative about using new language features in the Standard Library, and it was easier to just add an overload in `ranges`.

I suppose it makes sense to have `construct_at` in `ranges` for symmetry since there are `destroy_at` overloads in both namespaces.